interfaces/kmod/backend.go

```diff
@@ -67,9 +67,50 @@ func (b *Backend) Name() interfaces.SecuritySystem {
 	return "kmod"
 }
 
-// Setup creates a conf file with list of kernel modules required by given snap,
-// writes it in /etc/modules-load.d/ directory and immediately loads the modules
-// using /sbin/modprobe. The devMode is ignored.
+// setupModules creates a conf file with list of kernel modules required by
+// given snap, writes it in /etc/modules-load.d/ directory and immediately
+// loads the modules using /sbin/modprobe. The devMode is ignored.
+func (b *Backend) setupModules(snapInfo *snap.Info, spec *Specification) error {
+	content, modules := deriveContent(spec, snapInfo)
+	// synchronize the content with the filesystem
+	glob := interfaces.SecurityTagGlob(snapInfo.InstanceName())
+	dir := dirs.SnapKModModulesDir
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return fmt.Errorf("cannot create directory for kmod files %q: %s", dir, err)
+	}
+
+	changed, _, err := osutil.EnsureDirState(dirs.SnapKModModulesDir, glob, content)
+	if err != nil {
+		return err
+	}
+
+	if len(changed) > 0 {
+		b.loadModules(modules)
+	}
+	return nil
+}
+
+func (b *Backend) setupModprobe(snapInfo *snap.Info, spec *Specification) error {
```

Comment:
Nitpick, since setupModules above has a short description, maybe add one for setupModprobe as well?

Reply:
Added.